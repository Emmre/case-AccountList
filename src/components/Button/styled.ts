import styled from "styled-components";

interface InputProps {
  disabled?: string | any;
  color?: string | any;
  bg?: string | any;
  type?: string | any;
}

export const ButtonStyled = styled.button<InputProps>`
  border-radius: 12px;
  height: 56px;
  width: 100%;
  border: none;
  outline: none;
  background-color: ${(props) => (props.disabled ? "#BBC3CF" : props.bg)};
  color: ${(props) => (props.color ? props.color : "white")};
  border: 1px solid ${(props) => (props.color ? props.color : "transparent")};
`;
