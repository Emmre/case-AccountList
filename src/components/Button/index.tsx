import React, { FC, InputHTMLAttributes } from "react";

import { ButtonStyled } from "./styled";

interface IInputProps extends InputHTMLAttributes<HTMLButtonElement> {
  type: string;
  isDisabled: string;
  bg: string;
  color: string;
}

const Button: FC<IInputProps> = ({ children, type, isDisabled, ...rest }) => (
  <ButtonStyled
    type={type}
    disabled={!isDisabled?.length}
    value={isDisabled?.length}
    {...rest}
  >
    {children}
  </ButtonStyled>
);

export default Button;
