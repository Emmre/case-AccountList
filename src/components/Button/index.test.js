import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Button from "./index";

describe("Button component", () => {
  it("Should be rendering", () => {
    const { container } = render(<Button />);
    const button = container.querySelector("Button");
    expect(button).toBeInTheDocument();
  });

  it("Should match the snapshot", () => {
    const { asFragment } = render(<Button>Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
