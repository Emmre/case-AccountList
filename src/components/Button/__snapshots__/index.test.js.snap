// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL dYqtVo"
    disabled=""
  >
    Button
  </button>
</DocumentFragment>
`;
